### ISSUE TITLE
Иногда расширение не срабатывает на странице аниме

### ISSUE THREAD

**Issue by Seryiza** [SebyakinAndrei/IHeardYouLikeLicense#8]
Иногда открываю страницу аниме (напр., [Overlord](https://shikimori.org/animes/37675-overlord-iii)) и расширение не срабатывает.

![image](https://user-images.githubusercontent.com/2355735/44949956-e88a5d00-ae87-11e8-8210-a2c90c3593d2.png)

Консоль со страницы:
![image](https://user-images.githubusercontent.com/2355735/44949947-b0831a00-ae87-11e8-98f9-7a5c2f6c3a49.png)

Консоль из самого расширения:
![image](https://user-images.githubusercontent.com/2355735/44949952-c1cc2680-ae87-11e8-9730-746373fa659f.png)

**Comment by SebyakinAndrei** [comment]
Не срабатывает оно не из-за ошибок в консоли, а из-за того, что какие-то мелкие элементы на странице долго не подгружаются, например реклама, которая блокируется, и браузер не триггерит событие полной загрузки страницы, после которого запускается скрипт. По-другому нельзя, поскольку есть шанс, что на странице не будут находиться все необходимые расширению dom элементы. Ивент частичной загрузки страницы, который триггерится после загрузки dom, а не всей страницы, есть только в api firefox'а, в других браузерах его нет.

**Comment by Seryiza** [comment]
> только в api firefox'а, в других браузерах его нет
Я не писал подобных расширений, но разве нельзя сделать какой-либо хак / другое решение?

Например, как это делают в user scripts ([как это](https://openuserjs.org/scripts/ImoutoChan/Shiki_Rating/source)).

Потому что иначе от такой нестабильности и воли случая становится грустно.

**Comment by SebyakinAndrei** [comment]
Да, клевая тема. Пока что идея такая: иметь список с аттрибутами элементов, по которым мы их будем распознавать, далее при срабатывании observer'а сверяем эти данные с данными добавленных нодов, когда понимаем, что все необходимые ноды добавлены, запускаем скрипт.
Реализовать я ее в скором времени не смогу, но было бы неплохо такое запилить. Хотя, возможно, что подобный код может замедлить работу страниц.

**Comment by wakirin** [comment]
как насчет `DOMContentLoaded` ?
